targets/unity-v2/make.js

```diff
@@ -378,24 +417,34 @@ function getAuthParams(apiCall) {
     return "AuthType.None";
 }
 
-function getRequestActions(tabbing, apiCall) {
-    if (apiCall.name === "GetEntityToken")
+function getRequestActions(tabbing, apiCall, isApiInstance = false) {
+    if (apiCall.name === "GetEntityToken" && isApiInstance === false)
         return tabbing + "AuthType authType = AuthType.None;\n" +
             "#if !DISABLE_PLAYFABCLIENT_API\n" +
             tabbing + "if (authType == AuthType.None && PlayFabClientAPI.IsClientLoggedIn())\n" +
             tabbing + "    authType = AuthType.LoginSession;\n" +
             "#endif\n" +
             "#if ENABLE_PLAYFABSERVER_API || ENABLE_PLAYFABADMIN_API || UNITY_EDITOR\n" +
-            tabbing + "if (authType == AuthType.None && !string.IsNullOrEmpty(PlayFabSettings.DeveloperSecretKey))\n" +
+            tabbing + "if (authType == AuthType.None && !string.IsNullOrEmpty(request.AuthenticationContext.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey))\n" +
             tabbing + "    authType = AuthType.DevSecretKey;\n" +
             "#endif\n";
+	if (apiCall.name === "GetEntityToken" && isApiInstance === true)
+        return tabbing + "AuthType authType = AuthType.None;\n" +
+            "#if !DISABLE_PLAYFABCLIENT_API\n" +
+            tabbing + "if (authType == AuthType.None && authenticationContext.IsClientLoggedIn())\n" +
+            tabbing + "    authType = AuthType.LoginSession;\n" +
+            "#endif\n" +
+            "#if ENABLE_PLAYFABSERVER_API || ENABLE_PLAYFABADMIN_API || UNITY_EDITOR\n" +
+            tabbing + "if (authType == AuthType.None && !string.IsNullOrEmpty(request.AuthenticationContext.DeveloperSecretKey ?? (authenticationContext.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey)))\n" +
+            tabbing + "    authType = AuthType.DevSecretKey;\n" +
+            "#endif\n";		
 
     if (apiCall.result === "LoginResult" || apiCall.request === "RegisterPlayFabUserRequest")
         return tabbing + "request.TitleId = request.TitleId ?? PlayFabSettings.TitleId;\n";
-    if (apiCall.auth === "SessionTicket")
+    if (apiCall.auth === "SessionTicket" && isApiInstance === false)
```

Comment:
>SessionTicket [](start = 26, length = 13)

Ok, you are right: for this:
if (apiCall.auth === "SessionTicket") block let's not make a separation by isApiInstance as they will be the same. Just keep one single if (apiCall.auth === "SessionTicket") block which will have this:
return tabbing + "if (!IsClientLoggedIn()) throw new PlayFabException(PlayFabExceptionCode.NotLoggedIn,\"Must be logged in to call this method\");\n";

Both static and instance API classes have IsClientLoggedIn method so it should work.

Reply:
Since you showed me that we can't simply call IsClientLoggedIn everywhere, we may have to keep the separation by isApiInstance. Please use direct checks of ClientSessionTicked fields here instead of callign IsClientLogged in.

---
In reply to: [262695780](https://github.com/PlayFab/SDKGenerator/pull/420#discussion_r262695780) [](ancestors = 262695780)

Reply:
We can use it there because it is for session ticket and it is used for ClientApi i think. Am i wrong?

Reply:
I think this check is used for any API call that uses "Session Ticket" authentication (instead of, for example, "Secret Key" authentication). For "Session Ticket" authentication a ClientSessionTicket value will be used, which is why i think we can simply check it directly, instead of calling IsClientLoggedIn (which is doing exactly that - it checks if client session ticket value was set).

---
In reply to: [262704021](https://github.com/PlayFab/SDKGenerator/pull/420#discussion_r262704021) [](ancestors = 262704021)

Reply:
I commit it with single if block please let me know if i should revert it to double if blocks

Reply:
sorry there is a little confusion for me :) as a final decision what should I do

Reply:
Give me a moment, please. I will write it :)

---
In reply to: [262709936](https://github.com/PlayFab/SDKGenerator/pull/420#discussion_r262709936) [](ancestors = 262709936)

Reply:
Yeah it will be double blocks, see my other comment with direct replacement code

---
In reply to: [262709039](https://github.com/PlayFab/SDKGenerator/pull/420#discussion_r262709039) [](ancestors = 262709039)